tests/commands/test_store_commands.py

```diff
@@ -893,6 +893,22 @@ def test_upload_charm_with_init_template_todo_token(tmp_path, config):
         UploadCommand(config).run(args)
 
 
+def test_upload_with_different_name_than_in_metadata(emitter, store_mock, config, tmp_path):
+    """Simple upload, success result."""
```

Comment:
This docstring is misleading, adapt it to this test (explain the case that is being tested).